### PR TITLE
Sample integration for DS and Default dashboard feature

### PR DESCRIPTION
**Pull request by NipunaMarcus**
Also issue with newly created dashboard direct user for layout selection when browser refreshed on designer view of that dashboard.

### CODE REVIEW

**Comment by NipunaMarcus** [comment]
This PR fixes following jiras:
1. https://wso2.org/jira/browse/UES-637
2. https://wso2.org/jira/browse/UES-631

File: apps/portal/controllers/apis/dashboards.jag [hunk 1]
```diff
@@ -62,15 +62,15 @@
...
                 dashboard.isUserCustom = true;
             }
 
             if ((!user || !utils.allowed(user.roles, configs.designers) || !utils.allowed(user.roles,
-            editors)) && !(utils.allowed(user.roles, viewers) && dashboard.isUserCustom)) {
+                            editors)) && !(utils.allowed(user.roles, viewers) && dashboard.isUserCustom)) {
                 response.sendError(401, JSON.stringify({
-                error: 'unauthorized to access the dashboard'
-            }));
+                    error: 'unauthorized to access the dashboard'
```
Comment (sinthuja) [inline apps/portal/controllers/apis/dashboards.jag]: Start the message with upper case.

File: apps/portal/controllers/logout.jag [hunk 2]
```diff
@@ -1,22 +1,18 @@
 <%
```
Comment (sinthuja) [inline apps/portal/controllers/logout.jag]: Add licence header.

File: apps/portal/js/dashboard-deployer.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                log.info("Deleting existing dashboard by same id: " + dashboardJson.id);
+                dashboard.remove(dashboardJson.id);
+
+                log.info("Deploying the dashboard: " + dashboardJson.title);
+                dashboard.create(dashboardJson);
+            }
+        }
+    }
+} catch (Exception) {
+    log.error("Required Files for auto Dashboard Deployer are missing");
```
Comment (sinthuja) [inline apps/portal/js/dashboard-deployer.js]: Need to add more context on the actual exception that's thrown rather swallowing it.

File: apps/portal/js/designer.js [hunk 4]
```diff
@@ -1934,34 +1945,34 @@ $(function () {
...
             if (file.size == 0) {
                 return;
             }
-            
+
             if (!new RegExp('^image/', 'i').test(file.type)) {
                 return;
             }
-            
+
```
Comment (sinthuja) [inline apps/portal/js/designer.js]: Remove all unnecessary new lines.

File: apps/portal/theme/templates/create-gadget.jag [hunk 5]
```diff
@@ -21,8 +21,8 @@
         </div>
         <div class="collapse navbar-collapse ues-context-menu-actions">
             <ul class="nav navbar-nav">
-                <li class="active"><a href="<%=tenantedUrlPrefix%>./">Dashboards</a></li>
-                <li><a href="<%=tenantedUrlPrefix%>./">
+                <li class="active"><a href="<%=tenantedUrlPrefix%>./dashboards">Dashboards</a></li>
```
Comment (sinthuja) [inline apps/portal/theme/templates/create-gadget.jag]: Use locales for Dashboard.

File: apps/portal/theme/templates/dashboard.jag [hunk 6]
```diff
@@ -103,7 +107,7 @@
                                      <%= i18n.localize("personalize")%>
                                 </a>
                             </li>
-                        <% } %>
+                        <% }}%>
```
Comment (sinthuja) [inline apps/portal/theme/templates/dashboard.jag]: Formatting issue.

File: apps/portal/controllers/apis/dashboards.jag (hunk 1 above)
Comment (NipunaMarcus) [inline apps/portal/controllers/apis/dashboards.jag]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/173

File: apps/portal/controllers/logout.jag (hunk 2 above)
Comment (NipunaMarcus) [inline apps/portal/controllers/logout.jag]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/173

File: apps/portal/js/dashboard-deployer.js (hunk 3 above)
Comment (NipunaMarcus) [inline apps/portal/js/dashboard-deployer.js]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/173

File: apps/portal/js/designer.js (hunk 4 above)
Comment (NipunaMarcus) [inline apps/portal/js/designer.js]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/173

File: apps/portal/theme/templates/create-gadget.jag (hunk 5 above)
Comment (NipunaMarcus) [inline apps/portal/theme/templates/create-gadget.jag]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/173

File: apps/portal/theme/templates/dashboard.jag (hunk 6 above)
Comment (NipunaMarcus) [inline apps/portal/theme/templates/dashboard.jag]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/173

File: apps/portal/controllers/logout.jag [hunk 7]
```diff
@@ -1,22 +1,18 @@
...
-
-
-
+if (configs.authentication.activeMethod === 'sso') {
+    include('sso-logout.jag');
+} else {
+    var usr = require('/modules/user.js');
+    usr.logout();
+    response.sendRedirect(dest || tenantedUrlPrefix);
+    //var index = dest.indexOf('?isAnonView');
```
Comment (dakshika) [inline apps/portal/controllers/logout.jag]: better to remove commented lines

Comment (NipunaMarcus) [inline apps/portal/controllers/logout.jag]: Fixed in https://github.com/wso2/carbon-dashboards/pull/192

**Pull request merged**